### DISCUSSION TITLE
Conda vs HPC MOOSE

### DISCUSSION THREAD

**Discussion by ykvishal** [idaholab/moose#25519] [Q&A General] [answered]
Our HPC was updating its OS and some of the packages were not working properly. So, we were using conda version of MOOSE. Recently, I installed HPC version of MOOSE once all packages were working properly on the new HPC OS. I did a quick comparison of both versions of MOOSE and I got following results:

Test File: [grain_growth_2D_graintracker.i](https://github.com/idaholab/moose/blob/next/modules/phase_field/examples/grain_growth/grain_growth_2D_graintracker.i)

Case 1:
Number of steps: 100
Number of cores:64
Total time:
Conda MOOSE: 
 real         11m57.556s

 HPC MOOSE
real         4m59.171s

Case 2:
Next, I disabled adaptivity and uniform refine option and changed nx and ny from 44 to 1000. Everything else is same as Case 1. The performance is as follows
Total time:
Conda MOOSE
real         18m15.600s

HPC MOOSE 
real         12m17.326s

What is your opinion about these results? 

Also, I was running multiple simulation with conda version of MOOSE and I used shell script for this. Interestingly, some simulations crashed and core.* files were generated. I did not face this issue with the HPC version of MOOSE.

**Comment by GiudGiud** [comment, marked as answer]
Hello

your conda version of MOOSE is not using an MPI compiler that is optimizing or even compiled for your HPC infrastructure. The libmesh and petsc packages are compiled in an almost-architecture-agnostic manner

This is why you loose so much performance by using conda over HPC

wrt to adaptivity vs large grid, adaptivity is more efficient if it enables you to use less elements

Guillaume

**Comment by ykvishal** [comment]
I have another question related to HPC MOOSE performance. 
Is there any benefit if I use this petsc option `./scripts/update_and_rebuild_petsc.sh --with-64-bit-indices CXXOPTFLAGS=-g -O3 COPTFLAGS=-g -O3 FOPTFLAGS=-g -O3` ?

**Reply by GiudGiud** [reply]
-g is for debugging, -03 is for optimizing. These arent really aligned goals. The debug flags are already included for the debug build
The optimization flags are hard-coded at o2 in libmesh currently. PETSc at o3 could be faster, I have not tried
